src/Symfony/Component/DependencyInjection/Dumper/XmlDumper.php

```diff
@@ -298,6 +298,10 @@ private function convertParameters(array $parameters, $type, \DOMElement $parent
                 $element->setAttribute('type', 'expression');
                 $text = $this->document->createTextNode(self::phpToXml((string) $value));
                 $element->appendChild($text);
+            } elseif (\is_string($value) && !preg_match('/^[^\x00-\x08\x0B\x0E-\x1A\x1C-\x1F\x7F]*+$/u', $value)) {
```

Comment:
Do we still need the `u` modifier here?

Reply:
Yes, to ensure string is a valid UTF8 string for XML file.